internal/controller/storageclaim_controller.go

```diff
@@ -497,6 +501,10 @@ func (r *StorageClaimReconciler) getCephRBDStorageClass(data map[string]string)
 		Provisioner:          csi.GetRBDDriverName(),
 		Parameters:           data,
 	}
+
+	if r.storageClaim.Spec.EncryptionMethod != "" {
+		storageClass.ObjectMeta.Annotations[keyRotationAnnotation] = cronScheduleWeekly
```

Comment:
1. ObjectMeta is an embedded struct, you don't need to explicitly specify it
2. You cannot be sure that the annotations map is not nil. you first need to check it

```suggestion
		if storageClass.Annotations == nil {
			storageClass.Annotations = map[string]string{}
		}
		storageClass.Annotations[keyRotationAnnotation] = cronScheduleWeekly
```

But you don't need to be explicit here because we already have a util that implements the above snippet
```suggestion
		utils.AddAnnotation(&storageClass, keyRotationAnnotation, cronScheduleWeekly)
```

Reply:
In this case, annotations are explicitly set at line 493 and if empty they are initialized first by `utils.AddAnnotation`. Am I missing something?

Reply:
@black-dragon74 As a general remake, please do not mark a conversation provided as review comments as resolved. 

The decision on resolution needs to be decided after input from the person who submitted the review. It makes it very hard to track on re-review what was addressed correctly and what still needs discussion/additional fixes if someone who is not the review owner resolves the comment.

When you fix an issue leave a "fixed" or "done" reply on the comment thread. If the reviewer is a maintainer they will then resolve after verifying, if the reviewer is a non-maintainer (which means they do not have permission to resolve comments on other contributors` PRs), they will leave their ack comment or a thumbs-up emoji to indicate that the fix is accepted then you can resolve